fv3core/stencils/xppm.py

```diff
@@ -244,12 +244,20 @@ class XPiecewiseParabolic:
     Fortran name is xppm
     """
 
-    def __init__(self, namelist, iord, jfirst, jlast):
-        grid = spec.grid
-        assert namelist.grid_type < 3
-        self._dxa = grid.dxa
-        origin = (grid.is_, jfirst, 0)
-        domain = (grid.nic + 1, jlast - jfirst + 1, grid.npz + 1)
+    def __init__(
+        self,
+        grid: GridIndexing,
+        dxa,
+        grid_type: int,
+        iord,
+        j_domain: Literal["compute", "full"],
```

Comment:
adding special strings can sometimes be more confusing than passing numbers. Also the fortran code is flexible to any jstart and jend. I'm on the fence, maybe we can keep it, but should consider implications for other modules that also pass through index numbers.

Reply:
I'll refactor this to an `origin` and `domain` argument, which is general and also gets around the indexing issue I ran into.

Reply:
Done.